0015-automatic-cluster-upgrades/0015-automatic-cluster-upgrades.md

```diff
@@ -0,0 +1,151 @@
+# RFC 0015 - Automatic workload cluster upgrades
+
+## User stories
+
+- As a Giant Swarm customer, and also as Giant Swarm, we would like to have workload clusters running in the latest version automatically, and not worry about patching security vulnerabilities or cluster components running out of updates (e. g. Kubernetes release end of life).
+
+- As a customer admin, I would like to define maintenance windows to ensure workload clusters are only upgraded within these hours.
```

Comment:
I'd like to explicitly call it "upgrade window". Calling it "maintenance window" suggests that the window is defined for other purposes, such as manual maintenance, while in fact this RFC should be focused on upgrades.

```suggestion
- As a customer admin, I would like to define upgrade windows, to ensure that upgrades happen when disruption is most permittable and impact on production traffic is minimal.
```

If you accept this suggestion, please also update other occurrences of "maintenance window".